plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -627,10 +627,14 @@ func (c *CiscoTelemetryMDT) parseClassAttributeField(grouper *metric.SeriesGroup
 	for _, subfield := range nxAttributes.Fields {
 		if subfield.Name == "dn" {
 			tags["dn"] = decodeTag(subfield)
-		} else {
-			c.parseContentField(grouper, subfield, "", encodingPath, tags, timestamp)
+			break
 		}
 	}
+
+	for _, subfield := range nxAttributes.Fields {
+		c.parseContentField(grouper, subfield, "", encodingPath, tags, timestamp)
+	}
+	delete(tags, "dn")
```

Comment:
What is the point in setting the `dn` tag first and then deleting it?

Reply:
We delete `dn` tag to ensure it is not carried over to the next group of fields.